tests/test_reed_solomon.py

```diff
@@ -90,6 +91,24 @@ def test_fft_encode(fft_encoding_test_cases):
         assert actual == encoded
 
 
+def test_auto_encode_fft_disabled(encoding_test_cases):
+    # Just check if some encoder is being picked
```

Comment:
I think we should validate that the correct type of the encoder gets picked for the values. If there's a bug which creeps up and for some reason we reverse them then we will never know and the code will perform worse.

Reply:
I can't think of a good way since the selection might platform dependent. Can you suggest a way to test this?

Reply:
[Mocking](https://stackoverflow.com/questions/16134281/python-mocking-a-function-from-an-imported-module) is the answer here--
```
from unittest.mock import patch
import psutil

@patch('psutil.cpu_count')
def test(mocked_cpu_count):
    print(f"return value when you don't set return value: {psutil.cpu_count()}")
    mocked_cpu_count.return_value = 1
    print(f"return value after mocking: {psutil.cpu_count()}")
    mocked_cpu_count.return_value = 9000
    print(f"return value after mocking once more: {psutil.cpu_count()}")
    
print (f'normal cpu_count: {psutil.cpu_count()}')
test()
print (f'normal cpu_count after test: {psutil.cpu_count()}')
```
This outputs
```
normal cpu_count: 8
return value when you don't set return value: <MagicMock name='cpu_count()' id='140643940895544'>
return value after mocking: 1
return value after mocking once more: 9000
normal cpu_count after test: 8
```

Reply:
That's pretty cool. I'll try rewriting the test with this.